### PR TITLE
pd-ctl: make `TestShowKeyspaceGroupPrimary` stable

### LINKED ISSUE #6783: `TestShowKeyspaceGroupPrimary` failed
### Which jobs are failing
```
2023-07-11T02:48:26.8749960Z --- FAIL: TestShowKeyspaceGroupPrimary (27.68s)
2023-07-11T02:48:26.8750146Z     keyspace_group_test.go:566: 
2023-07-11T02:48:26.8750605Z         	Error Trace:	/home/runner/work/pd/pd/tests/pdctl/keyspace/keyspace_group_test.go:566
2023-07-11T02:48:26.8751264Z         	            				/opt/hostedtoolcache/go/1.20.1/x64/src/runtime/asm_amd64.s:1598
2023-07-11T02:48:26.8751508Z         	Error:      	Received unexpected error:
2023-07-11T02:48:26.8752123Z         	            	invalid character 'F' looking for beginning of value
2023-07-11T02:48:26.8752392Z         	Test:       	TestShowKeyspaceGroupPrimary
2023-07-11T02:48:26.8752545Z     testutil.go:63: 
2023-07-11T02:48:26.8752944Z         	Error Trace:	/home/runner/work/pd/pd/pkg/utils/testutil/testutil.go:63
2023-07-11T02:48:26.8753732Z         	            				/home/runner/work/pd/pd/tests/pdctl/keyspace/keyspace_group_test.go:560
2023-07-11T02:48:26.8753980Z         	Error:      	Condition never satisfied
2023-07-11T02:48:26.8754340Z         	Test:       	TestShowKeyspaceGroupPrimary
```

### CI link
https://github.com/tikv/pd/actions/runs/5515219290/jobs/10055314959?pr=6774

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #6783

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/pdctl/keyspace/keyspace_group_test.go [hunk 1]
```diff
@@ -561,8 +561,13 @@ func TestShowKeyspaceGroupPrimary(t *testing.T) {
 		args := []string{"-u", pdAddr, "keyspace-group"}
 		output, err := pdctl.ExecuteCommand(cmd, append(args, "1")...)
 		re.NoError(err)
+		if strings.Contains(string(output), "Failed") {
+			// It may be failed when meets error, such as [PD:etcd:ErrEtcdTxnConflict]etcd transaction failed, conflicted and rolled back
+			fmt.Println(string(output))
```
Comment (rleungx) [inline tests/pdctl/keyspace/keyspace_group_test.go:566]: remove it?

**Review by rleungx — COMMENTED** [review]
The rest LGTM

File: tests/pdctl/keyspace/keyspace_group_test.go (hunk 1 above)
Comment (lhy1024) [reply tests/pdctl/keyspace/keyspace_group_test.go:566]: I'm concerned about encountering other errors that cause `testutil.Eventually` to keep retrying and eventually fail. Without log output, we may not be able to know what happened.

Comment (rleungx) [reply tests/pdctl/keyspace/keyspace_group_test.go:566]: re.NoErrorf(err, "output: %s", string(output))

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**